tests/bootstrap/main.go

```diff
@@ -0,0 +1,154 @@
+	}
+
+	if err := checkBootstrap(*dataDir, *avalanchegoPath, uint32(*networkID), *useDynamicPorts, *wipeDataDir, *stateSyncEnabled, *maxDuration); err != nil {
+		log.Fatalf("Failed to check bootstrap: %v\n", err)
+	}
+}
+
+func checkBootstrap(
+	dataDir string,
+	avalanchegoPath string,
+	networkID uint32,
+	useDynamicPorts bool,
+	wipeDataDir bool,
+	stateSyncEnabled bool,
+	maxDuration time.Duration,
+) error {
+	flags := tmpnet.DefaultLocalhostFlags()
+	flags.SetDefaults(tmpnet.FlagsMap{
+		config.HealthCheckFreqKey: "30s",
+		// Minimize logging overhead
+		config.LogDisplayLevelKey: logging.Off.String(),
+	})
+	if !useDynamicPorts {
+		flags.SetDefaults(tmpnet.FlagsMap{
+			config.HTTPPortKey:    config.DefaultHTTPPort,
+			config.StakingPortKey: config.DefaultStakingPort,
+		})
+	}
+
+	if wipeDataDir && len(dataDir) > 0 {
+		if err := os.RemoveAll(dataDir); err != nil {
+			return fmt.Errorf("failed to remove data dir: %w", err)
+		}
+	}
+
+	networkName := constants.NetworkName(networkID)
+	syncString := "full-sync"
+	if stateSyncEnabled {
+		syncString = "state-sync"
+	}
+
+	// Create a new single-node network that will bootstrap from the specified network
+	network := &tmpnet.Network{
+		UUID:         uuid.NewString(),
+		NetworkID:    networkID,
+		Owner:        fmt.Sprintf("bootstrap-test-%s-%s", networkName, syncString),
+		Nodes:        tmpnet.NewNodesOrPanic(1),
+		DefaultFlags: flags,
+		DefaultRuntimeConfig: tmpnet.NodeRuntimeConfig{
+			// TODO(marun) Rename AvalancheGoPath to AvalanchegoPath
+			AvalancheGoPath: avalanchegoPath,
+		},
+		ChainConfigs: map[string]tmpnet.FlagsMap{
+			"C": {
+				"state-sync-enabled": stateSyncEnabled,
+			},
+		},
+	}
+
+	if err := network.Create(dataDir); err != nil {
+		return fmt.Errorf("failed to create network: %w", err)
+	}
+	node := network.Nodes[0]
+
+	log.Printf("Starting node in path %s (UUID: %s)\n", network.Dir, network.UUID)
+
+	ctx, cancel := context.WithTimeout(context.Background(), tmpnet.DefaultNetworkTimeout)
+	defer cancel()
+	if err := network.StartNode(ctx, os.Stdout, node); err != nil {
+		return fmt.Errorf("failed to start node: %w", err)
+	}
+	defer func() {
+		ctx, cancel := context.WithTimeout(context.Background(), tmpnet.DefaultNetworkTimeout)
+		defer cancel()
+		if err := node.Stop(ctx); err != nil {
+			log.Printf("Failed to stop node: %v\n", err)
+		}
+	}()
+
+	log.Printf("Metrics: %s\n", tmpnet.DefaultMetricsLink(network.UUID, time.Now()))
+
+	log.Print("Waiting for node to indicate bootstrap complete by reporting healthy\n")
+
+	// Avoid checking too often to prevent log spam
+	healthCheckInterval := 1 * time.Minute
+
+	ctx, cancel = context.WithTimeout(context.Background(), maxDuration)
+	defer cancel()
+	if err := tmpnet.WaitForHealthyWithInterval(ctx, node, healthCheckInterval); err != nil {
+		return fmt.Errorf("node failed to become healthy before timeout: %w", err)
+	}
+
+	log.Print("Bootstrap completed successfully!\n")
```

Comment:
q: thinking alond here,  I wonder if we should also test that the node can keep up with traffic once it goes out of bootstrap? Like run it some time after bootstrap completes and check that it does not fall behind too much.

Reply:
That seems out of scope